packages/iris-grid/src/IrisGridTableModelTemplate.ts

```diff
@@ -1617,15 +1617,18 @@ class IrisGridTableModelTemplate<
     ) {
       return false;
     }
-    return !this.isKeyColumn(modelIndex);
+    return !this.isKeyColumn(this.columns[modelIndex]);
   }
 
   isColumnSortable(modelIndex: ModelIndex): boolean {
     return this.columns[modelIndex].isSortable ?? true;
   }
 
-  isKeyColumn(x: ModelIndex): boolean {
-    return x < (this.inputTable?.keyColumns.length ?? 0);
+  isKeyColumn(column: DhType.Column): boolean {
+    if (column == null) {
+      return false;
+    }
+    return this.keyColumnSet.has(column.name);
```

Comment:
We shouldn't need to pass in the column here, just pass in the `ModelIndex` as we were before.
```typescript
isKeyColumn(x: ModelIndex): boolean {
  return this.keyColumnSet.has(this.columns[modelIndex].name);
}
```
Then you shouldn't need to change the calls to `isKeyColumn` elsewhere.